File: HDAVPlayer/DesignModels/MVP/MVPViewController.m
//
//  MVPViewController.m
//  HDAVPlayer
//
//  Created by 崔玉冠 on 2018/3/28.
//  Copyright © 2018年 崔玉冠. All rights reserved.
//

#import "MVPViewController.h"
#import "Presenter.h"
#import "MVPModel.h"
#import "MVPView.h"

@interface MVPViewController ()

@property (nonatomic, strong) Presenter *presenter;

@property (nonatomic, strong) MVPModel *mvpModel;

@property (nonatomic, strong) MVPView *mvpView;

@end

@implementation MVPViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    self.presenter = [[Presenter alloc] init];
    
    self.mvpView = [[MVPView alloc] init];
    self.mvpView.frame = self.view.frame;
    [self.view addSubview:self.mvpView];
    
    self.mvpModel = [[MVPModel alloc] init];
    self.mvpModel.content = @"line 0";
    
    self.presenter.mvpModel = self.mvpModel;
    self.presenter.mvpView = self.mvpView;
    [self.presenter printOpreate];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

@end

File: HDAVPlayer/DesignModels/Mvp/MvpImplement/MvpViewController.swift
//
//  MvpViewController.swift
//  HDAVPlayer
//
//  Created by 崔玉冠 on 2018/2/17.
//  Copyright © 2018年 崔玉冠. All rights reserved.
//

import UIKit

class MvpViewController: UIViewController, LoginView {
    
    private let presenter = LoginPresenter()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        presenter.attachView(view: self)
        // Do any additional setup after loading the view.
    }
    
    func onLoginResult(data: NSObject) {
        print("回到了UI层")
    }
    
    @IBAction func loginRequest(_ sender: Any) {
        self.presenter.login(name: "cuiyuguan", pwd: "cyg08212611")
    }
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        presenter.detachView(view: self)
        // Dispose of any resources that can be recreated.
    }
    
}

